### ISSUE TITLE
[GlobalStep] Rotating the device while taking a picture causes it not to be successfully uploaded

### ISSUE THREAD

**Issue by wptester9845** [wordpress-mobile/WordPress-Android#11296]
### Description
When adding a Photo to an Image block, if the user rotates the device while capturing the photo, it will fail to be uploaded and the error message “The media item couldn’t be retrieved” is displayed.

### Reproduction Rate
4/4 100%

### Expected behavior
The photo should be uploaded succesfully.

### Actual behavior
The error message “The media item couldn’t be retrieved” is displayed and the photo isn’t uploaded.

### Steps to reproduce the behavior
1. Install WordPress 14.2
2. Log in to an account.
3. Press the “Add new” button.
4. Select the “Blog post” option.
5. Add an Image block.
6. Press “Add Image”.
7. Select the “Take a photo” option.
8. Rotate the device to switch from Landscape mode to Portrait mode or vice-versa.
9. Capture the photo and confirm.
##### Tested on the following
Google Pixel C (8.1.0)
Huawei P20 Pro (9.0)

#### Please see the attached video and log for more information
[AndroidCaptureRotate.zip](https://github.com/wordpress-mobile/WordPress-Android/files/4198037/AndroidCaptureRotate.zip)
[AndroidCaptureRotateLog.txt](https://github.com/wordpress-mobile/WordPress-Android/files/4198038/AndroidCaptureRotateLog.txt)

### Submitted by:
Luis Pimenta

**Comment by jhnstn** [comment]
Issue still happens for me on Pixel 5 (Android 11) WPAndroid 17.0-rc-2

**Comment by zwarm** [comment]
Retested on:
Pixel 5, Android 12, WP app 19.1.1
Samsung S20, Android 12, WP 19.1.1 

Did not experience a crash. This _may_ have been fixed by the introduction of the new `GutenberyDialogFragment` in  this [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/14503)

@thehenrybyrd - I'm out of test devices, do you want to give this a run through on any that you have or shall I close it?

**Comment by thehenrybyrd** [comment]
Thanks for the ping @zwarm!
I tried this out on my Samsung Galaxy S21, and wasn't expecting a different result than you got, but I was still able to repro.
Turning the device after tapping "Take photo", but before snapping the photo, and then tapping "ok" _before_ turning the device back results in the following error:  
"Media upload failed"
"The specified path is a directory instead of a media file"

For example:
1. Start in Portrait mode.
2. Add an Image block and tap "Take a Photo" from the Choose image options.
3. Once the camera app has loaded, turn the device to Landscape mode.
4. Tap the on-screen button to take the photo, still in Landscape.
5. Tap the on-screen "OK" button to accept the photo, still in Landscape.
6. Note the error appears, and the image block displays "Failed to insert media. Please tap for options".

Interestingly, if I turn the device back to Portrait mode before tapping "OK" (between steps 4 and 5), the error does not occur.

https://user-images.githubusercontent.com/15107387/154442709-226f2dd7-5972-484c-95a5-0964d20ba508.mp4

**Comment by mkevins** [comment]
I have reproduced this with the [steps described here](https://github.com/wordpress-mobile/WordPress-Android/issues/11296#issuecomment-1042726664). I also tried the same steps within the media library, and did not encounter the issue there.

Investigating and debugging this a bit, I discovered that the media capture path was an empty string (`""` ) after following the steps to reproduce the issue. This information is stored in a callback as a member within `EditPostActivity` (`mMediaCapturePath`), but this is lost when the activity is recreated.

We can save this string and restore it after orientation changes (i.e. when the activity is destroyed), which resolves the issue. I will open a PR with a fix soon.
